### PR TITLE
Changes for cross-tenant integration

### PR DESCRIPTION
**Pull request by diyor-uptycs**
New changes do the following:
- Create a service principal for multi-tenant Uptycs App instead of app registration
- Create role assignments for the service principal
- Add the service principal as a Reader to the Subscription IAM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,6 +68,13 @@ module "iam-config" {
 
   # modify as you need
   resource_prefix = "uptycs-cloudquery-integration-123"
```
Comment (gbonthu) [inline README.md:70]:
```suggestion
  resource_prefix = "uptycs-integration-123"
```

**Comment by gbonthu** [comment]
@diyor-uptycs Please add the list of test-cases you tested (or could not test)?

File: README.md (hunk 1 above)
Comment (diyor-uptycs) [reply README.md:70]: Done

**Comment by diyor-uptycs** [comment]
@gbonthu We tested the following test cases:

1. Create new Tenant Integration
2. Create new Subscription Integration
3. Create new Subscription integration in the same tenant where service principal already exists
4. Migrate existing Tenant Integration
5. Migrate existing Subscription Integration

Here is the documentation including details of each test case:
https://uptycsjira.atlassian.net/wiki/spaces/CLD/pages/2661318930/Azure+Integration+Instructions+for+QA

File: README.md [hunk 2]
```diff
@@ -97,7 +106,7 @@ output "subscription_name" {
 | subscription_name | Name of the the azure subscription  |
 
 ```sh
-$ terraform init
+$ terraform init --upgrade
 $ terraform plan # Please verify before applying
 $ terraform apply
 # Once terraform is applied successfully, it will create "client_credentials.json" file.
```
Comment (llukalapu-uptycs) [inline README.md]: Can you modify this line. client_credentials.json will not be created right

File: README.md [hunk 3]
```diff
@@ -64,10 +64,17 @@ To execute the Terraform script:
 
 ```
 module "iam-config" {
-  source     = "github.com/uptycslabs/terraform-azure-iam-config"
+  source     = "github.com/uptycslabs/terraform-azurearm-ad-config"
```
Comment (llukalapu-uptycs) [inline README.md]: github.com/uptycslabs/terraform-azurerm-ad-config

Comment (diyor-uptycs) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (diyor-uptycs) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (llukalapu-uptycs) [reply README.md]: @diyor-uptycs  there is typo. Can you fixt it

**Review by djuneja-uptycs — APPROVED** [review]

**Pull request merged**